File: c#/src/Docmost.Sdk/Services/CloudApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class CloudApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> CreateCloudWorkspaceAsync(CreateCloudWorkspaceDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/create", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetJoinedWorkspacesAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/joined", null, cancellationToken).ConfigureAwait(false);
    }

}

File: typescript/src/services/cloud-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { CreateCloudWorkspaceDto } from '../models/index.js';


export class CloudApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async createCloudWorkspace(request: CreateCloudWorkspaceDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/create`, request);
  }

  async getJoinedWorkspaces(): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/joined`, undefined);
  }

}
